src/Analysis/Engine/Impl/AnalysisUnit.cs

```diff
@@ -478,14 +478,9 @@ public ComprehensionAnalysisUnit(Comprehension node, PythonAst parent, Interpret
 
         internal override void AnalyzeWorker(DDG ddg, CancellationToken cancel) {
             var comp = (Comprehension)Ast;
-            var forComp = comp.Iterators[0] as ComprehensionFor;
 
-            if (forComp != null) {
-                // evaluate the 1st iterator in the outer scope
-                ddg.Scope = _outerUnit.InterpreterScope;
+            if (comp.Iterators[0] is ComprehensionFor forComp) {
```

Comment:
Is it guaranteed that `comp.Iterators` has at least one item?

Reply:
Presumably. The original code picked out `[0]` unconditionally too.

Reply:
Yeah, I think it should always have at least one thing, but we may need to think about what happens when there's _more_ than one thing. For example, "fun" list comprehension to flatten a 2D list is:

```python
l = [[1, 2], [3, 4]]
f = [i for l2 in l for i in l2]
```

At the point of this call, `comp.Iterators` contains two things. It's probably a better idea to iterate and assign each. I'll give it a test quick.

Reply:
Iterating or not doesn't seem to change anything, from what I can tell. With or without this scope fix, I can't hover over anything but `l` and get the right type.